libraries/api/discussion_helper.cpp

```diff
@@ -183,17 +183,36 @@ namespace golos { namespace api {
 
             if (d.allow_curation_rewards) {
                 if (d.total_vote_weight > 0) {
+                    auto auction_window_reward = max_rewards.value * c.auction_window_weight / total_weight;
+                    auto votes_after_auction_window_weight = total_weight - c.votes_in_auction_window_weight - c.auction_window_weight;
+
+                    auto auw_time = d.created + d.auction_window_size;
+
+
                     const auto &cvidx = db.get_index<comment_vote_index>().indices().get<by_comment_weight_voter>();
                     for (auto itr = cvidx.lower_bound(d.id); itr != cvidx.end() && itr->comment == d.id; ++itr) {
-                        auto claim = ((max_rewards.value * uint128_t(itr->weight)) / total_weight).to_uint64();
+                        uint64_t claim = ((max_rewards.value * weight) / total_weight).to_uint64();
+                        if (db.has_hardfork(STEEMIT_HARDFORK_0_19__898) && d.auction_window_reward_destination == protocol::to_curators) {
```

Comment:
Here should be checking on `d.auction_window_reward_destination == protocol::to_curators && itr->last_update >= auw_time` without checking on HF.

And next checking is not needed